pkg/cosign/kubernetes/webhook/validator.go

```diff
@@ -268,27 +268,22 @@ func ValidatePolicy(ctx context.Context, ref name.Reference, kc authn.Keychain,
 		// https://github.com/sigstore/cosign/issues/1651
 
 		switch {
-		case authority.Key != nil:
-			// Get the key from authority data
-			if authorityKeys, fieldErr := parseAuthorityKeys(ctx, authority.Key.Data); fieldErr != nil {
-				authorityErrors = append(authorityErrors, errors.Wrap(fieldErr, "failed to parse Key values"))
+		case authority.Key != nil && len(authority.Key.PublicKeys) > 0:
+			// TODO(vaikas): What should happen if there are multiple keys
+			// Is it even allowed? 'valid' returns success if any key
+			// matches.
+			// https://github.com/sigstore/cosign/issues/1652
+			sps, err := valid(ctx, ref, authority.Key.PublicKeys, remoteOpts...)
+			if err != nil {
+				authorityErrors = append(authorityErrors, errors.Wrap(err, "failed to validate keys"))
+				continue
 			} else {
-				// TODO(vaikas): What should happen if there are multiple keys
-				// Is it even allowed? 'valid' returns success if any key
-				// matches.
-				// https://github.com/sigstore/cosign/issues/1652
-				sps, err := valid(ctx, ref, authorityKeys, remoteOpts...)
-				if err != nil {
-					authorityErrors = append(authorityErrors, errors.Wrap(err, "failed to validate keys"))
-					continue
-				} else {
-					if len(sps) > 0 {
-						logging.FromContext(ctx).Debugf("validated signature for %s, got %d signatures", len(sps))
-						return sps, nil
-					}
-					logging.FromContext(ctx).Errorf("no validSignatures found for %s", ref.Name())
-					authorityErrors = append(authorityErrors, fmt.Errorf("no valid signatures found for %s", ref.Name()))
+				if len(sps) > 0 {
+					logging.FromContext(ctx).Debugf("validated signature for %s, got %d signatures", len(sps))
```

Comment:
i believe you're missing one argument here.

Reply:
Oh, there is another instance below as well, ill add it there as well